File: medxc-persistence/src/main/java/com/dxc/medxc/persistence/model/Appointment.java
/**
 *
 */
package com.dxc.medxc.persistence.model;

import java.sql.Timestamp;
import java.util.List;

/**
 * @author yyayya
 */
public interface Appointment {

    /**
     * @return Status of the appointment
     */
    Status getStatus();

    /**
     * @return Id of the appointment
     */
    int getId();

    /**
     * @return date of the appointment
     */
    Timestamp getDate();

    /**
     * @return Id of the patient
     */
    Patient getPIN();

    /**
     * @return Specialty of the appointment
     */
    Specialty getSpecialty();

    /**
     * @return Id of the doctor
     */
    Doctor getDocNum();

    /**
     * @return Ordered tests from the appointment
     */
    List<Test> getTests();

    /**
     * @return the record created for this appointment
     */
    Record getRecord();

    /**
     * @return the patient for whom this appointment is registered
     */
    Patient getPatient();

    /**
     * @return the patient that perform the visitation
     */
    Doctor getDoctor();
}

File: medxc-gui/src/app/model/appointment.ts
import { Doctor } from './doctor';
import { Patient } from './patient';
import { Record } from './record';

export class Appointment {
    constructor(date: Date, status: string, specialty: string, record: Record, doctor: Doctor, patient: Patient) {
        this.date = date;
        this.status = status;
        this.doctor = doctor;
        this.patient = patient;
        this.record = record;
        this.specialty = specialty;
    }
    date: Date;
    status: string;
    doctor: Doctor;
    patient: Patient;
    record: Record;
    specialty: string;

}
